spec/annotations/object-model.yaml

```diff
@@ -121,6 +121,17 @@ definitions:
       title: Element Reference
       ref: "#/definitions/ElementReference"
 
+  "@ObjectModel.tenantWideUniqueName":
+    type: string
+    description: |-
+      Unique technical name of the entity within the tenant / isolation context it is deployed to. 
+      This may be used as a hint for database table names and help to keep them short enough.
+
+      Once chosen the technical name ID MUST be kept stable (immutable).
```

Comment:
@swennemers : We agreed to add this annotation, but added the statement that this MUST be kept stable. Changing this would be an incompatible change.